app/src/main/java/org/android/go/sopt/presentation/sign/SignViewModel.kt

```diff
@@ -16,23 +18,33 @@ class SignViewModel(
 
     var userInput: UserInfo? = null
 
-    private var _isValidSign = MutableStateFlow(false)
+    private var _isValidSign = MutableStateFlow<Boolean?>(null)
     val isValidSign get() = _isValidSign.asStateFlow()
 
-    private var _isCompleteSign = MutableStateFlow(false)
+    private var _isCompleteSign = MutableStateFlow<Boolean?>(null)
     val isCompleteSign get() = _isCompleteSign.asStateFlow()
 
     private var _isAutoSignIn = MutableStateFlow(gsDataStore.isLogin)
     val isAutoSignIn = _isAutoSignIn.asStateFlow()
 
     fun isValid() {
-        if (inputId.value.length in 6..10 && inputPassword.value.length in 8..12)
-            _isValidSign.value = true
+        viewModelScope.launch {
+            if (inputId.value.length in 6..10 && inputPassword.value.length in 8..12)
+                _isValidSign.value = true
+            else {
+                _isValidSign.value = false
+            }
```

Comment:
```suggestion
                _isValidSign.value = inputId.value.length in 6..10 && inputPassword.value.length in 8..12
```
반드시 반영해야하는 건 아닌데 이렇게 바꿀수도 있다는 걸 아시면 좋을 것 같아용!